Signal/src/call/CallAudioService.swift

```diff
@@ -299,12 +309,22 @@ import AVFoundation
     private func setAudioSession(category: String,
                                  mode: String? = nil,
                                  options: AVAudioSessionCategoryOptions = AVAudioSessionCategoryOptions(rawValue: 0)) {
+        
+        let session = AVAudioSession.sharedInstance()
         do {
             if #available(iOS 10.0, *), let mode = mode {
-                try AVAudioSession.sharedInstance().setCategory(category, mode: mode, options: options)
+                if session.category == category, session.mode == mode, session.categoryOptions == options {
```

Comment:
Nice. I've also read that calling `setCategory()` is more expensive than you'd think, so that's a nice win too.